apps/communications/contacts/test/unit/views/form/form_controller_test.js

```diff
@@ -0,0 +1,121 @@
+/* global LazyLoader, Form, ContactsService, MockL10n, mozContact */
+/* global  Matcher, MatchService */
+
+'use strict';
+
+require('/shared/test/unit/mocks/mock_l10n.js');
+require('/shared/js/lazy_loader.js');
+
+requireApp('communications/contacts/services/contacts.js');
+requireApp('communications/contacts/js/match_service.js');
+require('/shared/js/text_normalizer.js');
+require('/shared/js/simple_phone_matcher.js');
+require('/shared/js/contacts/contacts_matcher.js');
+require('/shared/js/contacts/import/utilities/misc.js');
+
+requireApp('communications/contacts/views/form/js/form_controller.js');
+
+
+suite('FormController', function() {
+
+  var realMozL10n;
+  var testContact;
+  var stubLazyLoader;
+  var stubContactServiceSave;
+  suiteSetup(function() {
+    realMozL10n = navigator.mozL10n;
+    navigator.mozL10n = MockL10n;
+    Form.init();
+  });
+
+  suiteTeardown(function() {
+    navigator.mozL10n = realMozL10n;
+    realMozL10n = null;
+  });
+
+  function dispatchEvent() {
```

Comment:
nit: as this function is going to dispatch the save contact event, we should rename it to `dispatchSaveEvent` or something that reflects the specific use.